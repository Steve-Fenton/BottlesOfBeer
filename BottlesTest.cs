using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace BottlesOfBeer
{
    [TestClass]
    public class BottlesTest
    {
        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestTheFirstVerse()
        {
            Approvals.Verify(new Bottles().Verse(99));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestAnotherVerse()
        {
            Approvals.Verify(new Bottles().Verse(3));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestVerse2()
        {
            Approvals.Verify(new Bottles().Verse(2));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestVerse1()
        {
            Approvals.Verify(new Bottles().Verse(1));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestVerse0()
        {
            Approvals.Verify(new Bottles().Verse(0));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestACoupleOfVerses()
        {
            Approvals.Verify(new Bottles().Verses(99, 98));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestAFewVerses()
        {
            Approvals.Verify(new Bottles().Verses(2, 0));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestTheWholeSong()
        {
            Approvals.Verify(new Bottles().Song());
        }
    }
}