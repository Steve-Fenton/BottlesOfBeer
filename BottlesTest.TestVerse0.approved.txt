No more bottles of beer on the wall, no more bottles of beer.
Go to the store and buy some more, 99 bottles of beer on the wall.
